lib/src/core.c

```diff
@@ -523,15 +523,24 @@ static void* chain_sync(void *arg)
                 device->timeout++;
                 if (device->timeout >= CC_DEVICE_TIMEOUT)
                 {
-                    DEBUG_MSG("device timeout (device id: %i)\n", device->id);
+                    //maybe another device is blocking sync messages,
+                    //do not disconnect anything if input buffer is not empty
+                    if (sp_input_waiting(handle->sp) != 0)
```

Comment:
I suggest to put this line above, there is no point to keep repeating this check (which happens on `handle`) for each device available.
So instead, my suggestion is to check for this property right after the `usleep` call. If indeed there is "input waiting data", reset timeout for all devices.

Alternatively, store the result of this function in variable before the loop iteration.
But then, reset timeout to 0 independently of `timeout` already reaching the limit.